core/templates/services/svg-sanitizer.service.spec.ts

```diff
@@ -80,22 +47,128 @@ describe('SvgSanitizerService', () => {
   });
 
   it('should check for invalid base64 images', () => {
-    expect(svgSanitizerService.isValidBase64Svg(invalidBase64data)).toBe(false);
+    expect(svgSanitizerService.isBase64Svg(invalidBase64data)).toBe(false);
   });
 
-  it(
-    'should return null when malicious SVG is requested as SafeResourceUrl',
-    () => {
-      expect(svgSanitizerService.getTrustedSvgResourceUrl(maliciousSvg)).toBe(
-        null);
-    });
+  it('should return null when a invalid base64 SVG is requested as' +
+  'SafeResourceUrl',
+  () => {
+    expect(svgSanitizerService.getTrustedSvgResourceUrl(
+      invalidBase64data)).toBeNull();
+  });
 
   it(
     'should return SafeResourceUrl when a safe SVG is requested as' +
-    'SafeResourceUrl',
+      'SafeResourceUrl',
     () => {
-      expect(svgSanitizerService.getTrustedSvgResourceUrl(safeSvg)).toBe(
-        safeSvg);
+      const testCases = [
+        {
+          svgString: (
```

Comment:
It is hard to parse through the SVG string to figure out what is being tested. Should we consider splitting this into smaller tests with clear descriptions?

Reply:
Actually, the test cases have different arrangements for invalid tags and attributes that might occur in a SVG. I think changing the test name to `should return safeResourceUrl after removing invalid tags and attributes` might give a clearer idea of what we are testing.

Reply:
Perhaps you could add some comments to describe the key points of each test case.

Reply:
Done!

Reply:
I would suggest updating the comments to say what the namea of the tag/attr(s) are (you can just put them in parens). That will make it easier for readers of the code to validate them (in case they need to modify the test cases for some reason).

Reply:
Done!